askama_shared/src/parser.rs

```diff
@@ -1011,15 +1011,33 @@ fn block_comment<'a>(i: &'a [u8], s: &'a Syntax<'a>) -> IResult<&'a [u8], Node<'
     let mut p = tuple((
         |i| tag_comment_start(i, s),
         opt(tag("-")),
-        take_until(s.comment_end),
+        |mut i| {
+            let before_end: &[u8];
+            loop {
+                let (end, inner) = take_until(s.comment_end)(i)?;
+                let nested_start: IResult<&[u8], _> = take_until(s.comment_start)(i);
+
+                match nested_start {
+                    Ok((start, _)) if start.as_ptr() < end.as_ptr() => {
+                        i = block_comment(start, s)?.0;
+                    }
+                    _ => {
+                        i = end;
+                        before_end = inner;
+                        break;
+                    }
+                }
+            }
+            Ok((i, before_end))
+        },
         |i| tag_comment_end(i, s),
     ));
-    let (i, (_, pws, inner, _)) = p(i)?;
+    let (i, (_, pws, before_end, _)) = p(i)?;
```

Comment:
This seems like an unrelated change, let's pull it out into a separate commit. (I'm not convinced of its merits, TBH.)

Reply:
Previously `inner` contained the whole comment. Now it only contains the last part "before the end". So for `{# foo #}` then `inner` would contain ` foo `. However for `{# foo {# bar #} baz #}` it would only contain ` bar `.

So I felt it was more appropriate to rename it to `before_end` instead of `inner`. To avoid future confusion.

Reply:
Okay, let's call it `tail`.

Reply:
> However for `{# foo {# bar #} baz #}` it would only contain `bar`.

Did you mean it would only contain `baz`?

Reply:
Yes, of course. Typo 😅